File: Backend/Src/models/Message.js
const mongoose = require('mongoose');

const MessageSchema = new mongoose.Schema({

groupId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Group',
    required: true
},
senderId:{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Candidate',
    required: true
},
message: {
    type: String,
    required: true
},

timestamp: {
    type: Date,
    default: Date.now
}

},{ timestamps: true });
module.exports = mongoose.model('Message', MessageSchema);

File: Frontend/placement_prep/lib/models/message.dart
class Message {
  final String groupId; 
  final String senderId;
  final String message;
   
   Message({required this.groupId, required this.message, required this.senderId});
   factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
    message: json['message'],
      senderId: json['senderId'],
      groupId: json['groupId'],
    );
  }


}
